geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/Transformers.scala

```diff
@@ -142,8 +147,11 @@ object Transformers extends JavaTokenParsers {
   case class FieldLookup(n: String) extends Expr {
     var idx = -1
     override def eval(args: Any*)(implicit ctx: EvaluationContext): Any = {
-      if(idx == -1) idx = ctx.indexOf(n)
-      ctx.lookup(idx)
+      if (ctx.globalParams.isDefined && ctx.globalParams.get.isDefinedAt(n)) ctx.globalParams.get.getOrElse(n, null)
```

Comment:
maybe you could make this a match with unapply to avoid the .gets
ctx.globalParams match {
  case Some(params) if params.isDefinedAt(n) => params.getOrElse...
  case None => // logic from else
}